src/SpeedJumpIrm.sol

```diff
@@ -139,13 +141,14 @@ contract AdaptativeCurveIrm is IIrm {
                 startRateAtTarget.wMulDown(MathLib.wExp(linearAdaptation)).bound(MIN_RATE_AT_TARGET, MAX_RATE_AT_TARGET);
 
             // Then we compute the average rate over the period, with a Riemann sum.
-            int256 averageRateAtTarget;
+            // We omit the multiplication by the rectangle length because we would divide everything by the total length
+            // at the end, because we want to compute the average and not the integral.
+            int256 sum;
             int256 step = linearAdaptation / N_STEPS;
             for (int256 k = 1; k <= N_STEPS; k++) {
-                averageRateAtTarget += startRateAtTarget.wMulDown(MathLib.wExp(step * k)).bound(
-                    MIN_RATE_AT_TARGET, MAX_RATE_AT_TARGET
-                ) / N_STEPS;
+                sum += startRateAtTarget.wMulDown(MathLib.wExp(step * k)).bound(MIN_RATE_AT_TARGET, MAX_RATE_AT_TARGET);
```

Comment:
should we bound at the end only, instead?

Reply:
```
1.bound(0, 10) + 15.bound(0, 10) = 1 + 10 = 11
(1 + 15).bound(2*0,2*10) = 16.bound(0, 20) = 16
```

Reply:
Bounding at each step ensures that the average rate is between the start rate and the end rate, so I would keep the code as is

Reply:
I disagree: we know that `wMulDown` & `wExp` are increasing, so `startRateAtTarget <= startRateAtTarget.wMulDown(MathLib.wExp(step * k)) <= startRateAtTarget.wMulDown(linearAdaptation)`

So if you avg, you get `startRateAtTarget <= avgRateAtTarget <= startRateAtTarget.wMulDown(linearAdaptation)`

Then, given that `bound` is also increasing, it is guaranteed that `startRateAtTarget.bound() <= avgRateAtTarget.bound() <= endRateAtTarget`

Reply:
Smart, it seems to work. I have 2 more arguments for you to fight:
- bounding early ensures that the wExp upper value is not used. The result is used to be multiplied by a rate in the bounds (which we expect, and we have a test for it), and then it is bounded so we get back to a rate that is in the bounds. I think the suggestion would work in practice but it seems less clear
- right now it works as if the rate is really bounded by the `MAX_RATE_AT_TARGET`, whereas with the suggestion rates are allowed to temporarily be out of the bounds, which seems unexpected

Reply:
`startRateAtTarget <= avgRateAtTarget` is wrong no?

Reply:
Not true in all cases, just for `err >= 0`. For `err < 0` the inequalities are in the other direction but the same reasoning applies I think

Reply:
I'm in favor of bounding at each step because it gives the same behavior as if we were doing the average while the rate is bounded, which is not the case if we bound at the end